bootstrap4/.js/src/main/scala/io/udash/bootstrap/datepicker/UdashDatePicker.scala

```diff
@@ -531,4 +534,37 @@ object UdashDatePicker {
     def valueOf(): Double = js.native
   }
 
+  import org.scalajs.dom.raw.{MutationObserver, MutationObserverInit, MutationRecord}
+  import scala.collection.mutable.{Set => MSet}
+
+  private val datePickerSetupCallbacks = MSet.empty[(ComponentId, () => Unit)]
+
+  // When a date picker gets appended to a DOM, its options and listeners should be initialized once again. Thus, all
+  // nodes with mutated children are examined whether there is a date picker defined within a corresponding DOM
+  // subtree. It is also substantially possible for any date picker to be reported by more than one `MutationRecord`
+  // since the `subtree` switch of `MutationObserverInit` is on, thus only the first update is considered for each
+  // picker (note `exists` call) to avoid double initialization. This mutation observer is turned off when there are no
+  // components registered to be watched.
+  private val datePickerMutationObserver =
+    new MutationObserver((records: js.Array[MutationRecord], _: MutationObserver) => {
+      val addedNodes = records.flatMap(record => for {i <- 0 until record.addedNodes.length} yield record.addedNodes(i))
+      datePickerSetupCallbacks.foreach { case (pickerId, callback) =>
+        if (addedNodes.exists {
+          case element: Element => element.querySelector(s"#$pickerId") != null
+          case _ => false
+        }) callback()
+      }
+    })
+
+  def registerSetupCallback(id: ComponentId, callback: () => Unit): Unit = {
+    datePickerSetupCallbacks += ((id, callback))
+    if (datePickerSetupCallbacks.isEmpty)
```

Comment:
I don't understand, is this ever called? You've just added a callback line above.

Reply:
That's the root cause for failing tests as well, nice catch 👍